lib/collection/src/collection.rs

```diff
@@ -1173,7 +1162,52 @@ impl Collection {
                     }
                     drop(read_shard_holder);
 
-                    // 2 - mark peer as inactive
+                    // 2 - promote local or remote to replica set if needed
+                    if *peer_id == self.this_peer_id {
+                        let mut write_shard_holder = self.shards_holder.write().await;
+                        let shard_to_insert = match write_shard_holder.take_shard(*shard_id) {
+                            Some(Shard::Local(local_shard)) => {
+                                let new_replica = ReplicaSetShard::build_from_local(
+                                    *shard_id,
+                                    self.this_peer_id,
+                                    local_shard,
+                                    self.notify_peer_failure_cb.clone(),
+                                )
+                                .await?;
+                                Shard::ReplicaSet(new_replica)
+                            }
+                            Some(Shard::Remote(remote_shard)) => {
+                                // TODO is that the best way to retrieve the current path?
+                                let shard_path = latest_shard_paths(&self.path, *shard_id)
```

Comment:
Seems so, @generall might know it better though

Reply:
looks alright